### PR TITLE
11.0 f31070c4c98055663bf83cdab4d75dbb8117a5f0 3

### PR DESCRIPTION
**Pull request by KolushovAlexandr**

### CODE REVIEW

File: pos_keyboard/doc/changelog.rst [hunk 1]
```diff
@@ -1,3 +1,14 @@
...
+`1.1.5`
+-------
+
+**Fix:** Function 'disconnect' not working in expected way
+
+`1.1.4`
+-------
+
+**Fix:** Incompatibility with some modules led to the action duplicating
+
```
[the remark this answers is not in the document]
Comment (Ramil-Mukhametzyanov) [reply pos_keyboard/doc/changelog.rst:11]: :+1: checked

File: pos_cashier_select/doc/changelog.rst [hunk 2]
```diff
@@ -1,3 +1,8 @@
+`1.0.2`
+-------
+
+**Fix:** Incompatibility with pos_scan_ref module
+
```
[the remark this answers is not in the document]
Comment (Ramil-Mukhametzyanov) [reply pos_cashier_select/doc/changelog.rst:5]: #976
@Ommo73 @em230418 @KolushovAlexandr 
How to test this?

File: pos_product_available_negative/doc/changelog.rst [hunk 3]
```diff
@@ -1,7 +1,7 @@
-.. _changelog:
+`1.0.2`
+-------
 
-Updates
-=======
+- **Fix:** Cashier was changed after giving the authorization of the sale
```
Comment (Ramil-Mukhametzyanov) [inline pos_product_available_negative/doc/changelog.rst:4]: #1040 
@ilmir-k @yelizariev @KolushovAlexandr 
How to test this?

File: pos_longpolling/doc/changelog.rst [hunk 4]
```diff
@@ -1,3 +1,8 @@
+`2.2.0`
+-------
+
+- **IMP:** *Autostart longpolling* field is hidden on the `POS config` form to prevent possible misconfigurations.
+
```
[the remark this answers is not in the document]
Comment (Ramil-Mukhametzyanov) [reply pos_longpolling/doc/changelog.rst:5]: :-1: I can not find this field

File: pos_multi_session_restaurant/doc/changelog.rst [hunk 5]
```diff
@@ -1,5 +1,11 @@
+`3.3.2`
+-------
+
+**Fix:** Error on starting a multisession without any floors
+
```
[the remark this answers is not in the document]
Comment (Ramil-Mukhametzyanov) [reply pos_multi_session_restaurant/doc/changelog.rst:5]: :-1: checked

File: pos_multi_session_sync/doc/changelog.rst [hunk 6]
```diff
@@ -1,3 +1,8 @@
+`1.0.7`
+-------
+- **Fix:** Updated 'update_revision_ID' message
+- **Fix:** Possible error with multiple multi sessions with the same multi session ID
+
```
[the remark this answers is not in the document]
Comment (Ramil-Mukhametzyanov) [reply pos_multi_session_sync/doc/changelog.rst:5]: :+1: checked

File: pos_product_available_negative/doc/changelog.rst (hunk 3 above)
Comment (KolushovAlexandr) [reply pos_product_available_negative/doc/changelog.rst:4]: When, for example, Demo is asking for permission to smth, and pos manager gives the permission, then the cashier was changed. But it shouldn't. That's the fix. So, change cashier to rightless demo, try to sell a product with negative quantity and when permission is given check that the cashier has not been changed from Demo to manager

File: pos_longpolling/doc/changelog.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (KolushovAlexandr) [reply pos_longpolling/doc/changelog.rst:5]: Yep it's hidden, should be :+1:  I think

File: pos_product_available_negative/doc/changelog.rst (hunk 3 above)
Comment (ilmir-k) [reply pos_product_available_negative/doc/changelog.rst:4]: Once a user who has authorization to confirm the order payment, click on  ``Confirm``, the cashier gets replaced by this authorized user

File: pos_cashier_select/doc/changelog.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ommo73) [reply pos_cashier_select/doc/changelog.rst:5]: @Ramil-Mukhametzyanov before this update, when installing both modules, they did not work properly and caused an error. So after that they just work properly.

File: pos_multi_session_restaurant/doc/changelog.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (yelizariev) [reply pos_multi_session_restaurant/doc/changelog.rst:5]: What is the problem? Do do you mean, that something has to be fixed?

File: pos_cashier_select/doc/changelog.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (KolushovAlexandr) [reply pos_cashier_select/doc/changelog.rst:5]: I think he meant scanning barcodes in a POS

File: pos_product_available_negative/doc/changelog.rst (hunk 3 above)
Comment (Ramil-Mukhametzyanov) [reply pos_product_available_negative/doc/changelog.rst:4]: :+1: checked

File: pos_longpolling/doc/changelog.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (Ramil-Mukhametzyanov) [reply pos_longpolling/doc/changelog.rst:5]: :+1: checked, fixed

File: pos_invoice_pay/doc/changelog.rst [hunk 7]
```diff
@@ -1,7 +1,12 @@
-`1.2.1`
+`1.2.2`
```
[the remark this answers is not in the document]
Comment (Ramil-Mukhametzyanov) [reply pos_invoice_pay/doc/changelog.rst:1]: :+1: checked

**Review by Ramil-Mukhametzyanov — APPROVED** [review]

File: pos_multi_session_restaurant/doc/changelog.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (Ramil-Mukhametzyanov) [reply pos_multi_session_restaurant/doc/changelog.rst:5]: :+1: Not the merge error

**Review by em230418 — CHANGES_REQUESTED** [review]
Сценарий tour_pos_invoice_pay не проходит, если через одуап запускать.
После Validate появляется вот это.
http://i.imgur.com/1lx3pmN.png

Для разворачивания выполнил это:
```
docker-compose run dodoo -- init --new-database pos1126 --demo -m account_invoicing,http_routing,account,sales_team,base_automation,mail,barcodes,base_setup,sale_management,pos_qr_scan,web_planner,stock,web_editor,point_of_sale,pos_restaurant,pos_disable_payment,analytic,pos_restaurant_base,resource,stock_account,pos_product_available,wechat,base,portal,web_tour,bus,pos_longpolling,product,web,pos_qr_payments,sale,pos_multi_session,qr_payments,pos_qr_show,decimal_precision,pos_pin,pos_cashier_select,pos_disable_payment_restaurant,pos_invoice_pay,pos_keyboard,pos_multi_session_restaurant,pos_multi_session_sync,pos_order_print_check,pos_partner_deselection,pos_product_available_negative,pos_product_lot,pos_scan_ref,pos_wechat
```

Для запуска тестов выполнил это:
```
docker-compose run dodoo -- init --new-database pos1126 --demo -m account_invoicing,http_routing,account,sales_team,base_automation,mail,barcodes,base_setup,sale_management,pos_qr_scan,web_planner,stock,web_editor,point_of_sale,pos_restaurant,pos_disable_payment,analytic,pos_restaurant_base,resource,stock_account,pos_product_available,wechat,base,portal,web_tour,bus,pos_longpolling,product,web,pos_qr_payments,sale,pos_multi_session,qr_payments,pos_qr_show,decimal_precision,pos_pin,pos_cashier_select,pos_disable_payment_restaurant,pos_invoice_pay,pos_keyboard,pos_multi_session_restaurant,pos_multi_session_sync,pos_order_print_check,pos_partner_deselection,pos_product_available_negative,pos_product_lot,pos_scan_ref,pos_wechat
```

Логи:
[pos1126.log](https://github.com/it-projects-llc/pos-addons/files/3770680/pos1126.log)

**Comment by em230418** [comment]
Линты надо пропавить. Все остальное - ок.

File: pos_invoice_pay/tests/test_pos_invoice.py [hunk 8]
```diff
@@ -39,6 +39,7 @@ def __init__(self):
 
             def create(self, param):
                 self.param = param
+                self.super()
```
Comment (em230418) [inline pos_invoice_pay/tests/test_pos_invoice.py]: Там нету такого метода. Предлагается просто сделать так:
```def create(self, param):  # pylint: disable=no-member```

**Review by em230418 — APPROVED** [review]

**Comment by em230418** [comment]
I approve to merge it now

**Pull request merged**